test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -2757,6 +2764,87 @@ UCS_TEST_P(test_ucp_sockaddr_protocols_diff_config,
 UCP_INSTANTIATE_CM_TEST_CASE(test_ucp_sockaddr_protocols_diff_config)
 
 
+class test_ucp_sockaddr_protocols_diff_net_devices :
+        public test_ucp_sockaddr_protocols
+{
+public:
+    void init()
+    {
+        if (is_self()) {
+            UCS_TEST_SKIP_R("self - same devices");
+        }
+
+        m_err_count = 0;
+
+        // Set number of Eager/RNDV lanes to be 1 to create only 1 UCT endpoint
+        modify_config("MAX_EAGER_LANES", "1");
+        modify_config("MAX_RNDV_LANES", "1");
+
+        modify_config("RNDV_THRESH", "0");
+
+        get_sockaddr();
+        test_base::init();
+    }
+
+    std::string find_net_device()
+    {
+        auto e = create_entity(); // temporary entity
+
+        std::string net_device;
+        ucp_rsc_index_t tl_idx;
+        for (tl_idx = 0; tl_idx < sender().ucph()->num_tls; ++tl_idx) {
+            uct_tl_resource_desc_t *tl_rsc = &e->ucph()->tl_rscs[tl_idx].tl_rsc;
+            if (tl_rsc->dev_type == UCT_DEVICE_TYPE_NET) {
+                if (m_test_addr.netdev_name().compare(tl_rsc->dev_name) == 0) {
+                    net_device = m_test_addr.netdev_name();
+                    goto out;
+                }
+            }
+        }
+
+out:
+        m_entities.remove(e);
+
+        if (net_device.empty()) {
+            UCS_TEST_SKIP_R("no suitable device found to match \"" +
+                            m_test_addr.netdev_name() + "\" CM iface");
+        }
+
+        return net_device;
+    }
+
+    void create_entities_and_connect()
+    {
+        std::string net_device = find_net_device();
+
+        {
+            modify_config("NET_DEVICES", net_device, FAIL_IF_NOT_EXIST);
```

Comment:
can you document what is this test doing ? maybe give example with specific device names

Reply:
also, found bug on a server when restricting `UCX_NET_DEVICES` for server instead of client